Fix issue #10

jkterry1 opened the pull request:
Fixes #10 with ~5 new lines of code and some refactoring. Did it because I need to deploy this to a few servers and this will solve problems.

jkterry1 committed:
.txt files in lists that return for code clarity

jkterry1 committed:
adds comments on built in stop word lists

jkterry1 committed:
fixes bug #10

justinkterry commented:
A bit of clarification on the motives and what it does: Installing  this via pip doesn't play nice with the included .txts, so this just provides a lot of ease of use for everyone. More importantly, if you're using this behind a webapp (ie me), and you want to easily distribute it across servers, then dealing with file location is a big problem because different server names. Also it inherently can't be cross platform. 

This solves that by putting the .txt files into made premade lists (it's easier to return, saves a little processing time if you're using this a lot on servers like me and has no serious downside) in separate files which are called if their name with no extension is put as the file location. The separate files are only imported on call by name so they don't degrade memory either, and including them by pip only adds a ~3kb to the package size.

jkterry1 committed:
Make flake8 stop complaining about no space at end of file

jkterry1 committed:
Make flake8 stop complaining about no space at end of file

jkterry1 committed:
Update SmartStopList.py

justinkterry commented:
The fail is over some tiny flake8 things that I can't track down easily. It all works well, you can fix them if you want to/can, its fine. The code has no visual anomalies to me.

mikeando commented on README.md:
```diff
@@ -11,11 +11,14 @@ The source code is released under the MIT License.
+    Rake.run(text);
+#for built in stoplists
+    import RAKE
+    Rake = RAKE.Rake(<keyword>); #current keywords are SmartStopList and FoxStopList
```
I'd wrap these in quotes, so its clearer to the user that the `<keyword>` is a string

~~~
RAKE.Rake(<keyword>); #current keywords are "SmartStopList" and "FoxStopList"
~~~

Otherwise as a user I'd be tempted to write

~~~
Rake = RAKE.Rake(SmartStopList)
~~~

Though arguably 

~~~
Rake = RAKE.Rake(RAKE.SmartStopList())
~~~

would be nice too?

jkterry1 replied:
In retrospect there's a much better way to do it as RAKE.SmartStopList() that I didn't think of. Changing now.

mikeando commented:
This looks like it would solve my issue nicely.

jkterry1 committed:
implement previously discussed fix and adds support for inputting lists, cleaned

jkterry1 committed:
updates readme to new method

jkterry1 committed:
typo sorry

jkterry1 committed:
another typo. I promise I speak english and proof read things. really.

justinkterry commented:
@mikeando better now? Also this way you can insert stop lists as actual lists, which I think is very nice.

mikeando commented on RAKE/RAKE.py:
```diff
@@ -56,6 +65,15 @@ def split_sentences(text):
     return sentences
 
 
+def build_stop_word_regex_list(stop_word_list): #static handles built in files only
```
You can remove this comment now.

Also you probably should change `build_stop_word_regex` to use this. It looks like it could be something like:

~~~
def build_stop_word_regex(stop_word_file_path):
  return build_stop_work_regex_list( load_stop_words(stop_word_file_path) )
~~~

jkterry1 replied:
I did one better :)

mikeando commented on README.md:
```diff
@@ -11,11 +11,18 @@ The source code is released under the MIT License.
+    Rake.run(text);
+#RAKE.SmartStopList() and RAKE.FoxStopList() return the expected lists as lists, they can be used as shown bellow
+    import RAKE
+    Rake = RAKE.Rake(RAKE.SmartStopList()); #takes list input
```
This markdown is not rendering correctly

jkterry1 replied:
it looks right to me, please explain?

mikeando replied:
Try the "view" link on the diff - https://github.com/jkterry1/python-rake/blob/d32efa9f758c6e39d5896cd6b2cc91406357b9b6/README.md You'll find that this section is mangled.

I think you want something closer to 

~~~
### Usage ###
for external .txt files

    import RAKE
    Rake = RAKE.Rake(<path_to_your_stopwords_file>); #take path as string datatype
    Rake.run(text);

for lists:

    import RAKE
    Rake = RAKE.Rake(<list>); #takes list input
    Rake.run(text);

`RAKE.SmartStopList()` and `RAKE.FoxStopList()` return the expected lists as lists, they can be used as shown bellow:

    import RAKE
    Rake = RAKE.Rake(RAKE.SmartStopList()); #takes list input
~~~

mikeando commented:
I like this even more. Nice one.

jkterry1 committed:
does mikeandos comment

removes unneeded code

jkterry1 committed:
pushing @mikeando s formatting fix

mikeando commented on README.md:
```diff
@@ -11,11 +11,22 @@ The source code is released under the MIT License.
+`RAKE.SmartStopList()` and `RAKE.FoxStopList()` return the expected lists as lists, they can be used as shown bellow:
+
+    import RAKE
+    Rake = RAKE.Rake(RAKE.SmartStopList()); #takes list input
```
I think trim the comment off the end of this line, as it adds nothing now.

mikeando commented:
I think I'm done finding tiny issues with this... it looks good to me, and IMO will go a long way towards making this usable directly after install.

jkterry1 committed:
removes unnecessary # case

isnt needed with new stopword list system and shouldnt be needed with normal lists

jkterry1 committed:
Update README.md

jkterry1 committed:
notes that only space separation is supported in .txt files

justinkterry commented:
@mikeando cool thanks. Can you think of anything else for it while its all fresh in my mind?

fabianvf commented on RAKE/FoxStopList.py:
```diff
@@ -0,0 +1,430 @@
+list=['a',
```
list is a reserved keyword in python, can be dangerous to collide with it. Recommend you rename it to `stopwords` or something just to avoid unwanted side-effects

jkterry1 replied:
done

fabianvf commented on RAKE/SmartStopList.py:
```diff
@@ -0,0 +1,576 @@
+list= ['a',
```
Same comments as above

jkterry1 replied:
done

fabianvf reviewed (changes requested):
I think you can go ahead and delete the text files as well, they don't serve much purpose with these lists in place. I like this change, other than the `list` keyword nit LGTM. Don't worry about the flake failures, I can make travis happy post merge.

fabianvf commented on RAKE/FoxStopList.py:
```diff
@@ -0,0 +1,430 @@
+ 'z']
+
+
+def words():
```
Why not just have the user import the list defined above?

fabianvf replied:
Never mind, makes sense since it allows them to define additional logic for creating the word list.

jkterry1 committed:
keyword fix 1

jkterry1 committed:
fix reserve word issue #2

jkterry1 committed:
remove unneeded

jkterry1 committed:
remove unneeded txt 2

jkterry1 committed:
this is a big security vulnerability dude

justinkterry commented:
@fabianvf did the things, also please make sure to not include the password to pypi in the future. I have to deploy this to servers so man in the middle attacks and all. this is issue #15 . I deleted it in my version so it's auto removed when you approve the pull request, but you need to add this to your git ignore and be careful about this in the future. I'll look at a smarter way to do issue #14 after lunch and see if one exists.

fabianvf commented on .travis.yml:
```diff
@@ -1,24 +0,0 @@
-language: python
```
Please leave this file in, the password is not actually leaked.

fabianvf replied:
https://docs.travis-ci.com/user/deployment/pypi/

jkterry1 committed:
replace .travis.yml, sorry

fabianvf commented:
haha, no need to apologize, without knowing about that specific travis feature it's an easy mistake to make, and definitely a good thing to call out.
